### ISSUE TITLE
Expose normalization functionality in EmailStr

### ISSUE THREAD

**Issue by tamird** [pydantic/pydantic#9451]
### Initial Checks

- [X] I have searched Google & GitHub for similar requests and couldn't find anything
- [X] I have read and followed [the docs](https://docs.pydantic.dev) and still think this feature is missing

### Description

`email-validator` provides a [normalized view](https://github.com/JoshData/python-email-validator/blob/5cf49cf87478a421df21ffeff9a1c87e30470e09/email_validator/exceptions_types.py#L31) of validated email addresses. It should be possible to get this from `EmailStr`.

This was mentioned in #9416 which was closed.

### Affected Components

- [ ] [Compatibility between releases](https://docs.pydantic.dev/changelog/)
- [X] [Data validation/parsing](https://docs.pydantic.dev/concepts/models/#basic-model-usage)
- [ ] [Data serialization](https://docs.pydantic.dev/concepts/serialization/) - `.model_dump()` and `.model_dump_json()`
- [ ] [JSON Schema](https://docs.pydantic.dev/concepts/json_schema/)
- [ ] [Dataclasses](https://docs.pydantic.dev/concepts/dataclasses/)
- [ ] [Model Config](https://docs.pydantic.dev/concepts/config/)
- [ ] [Field Types](https://docs.pydantic.dev/api/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://docs.pydantic.dev/concepts/validation_decorator/)
- [ ] [Generic Models](https://docs.pydantic.dev/concepts/models/#generic-models)
- [ ] [Other Model behaviour](https://docs.pydantic.dev/concepts/models/) - `model_construct()`, pickling, private attributes, ORM mode
- [X] [Plugins](https://docs.pydantic.dev/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by sydney-runkle** [comment]
@tamird,

Ah I see, thanks for elaborating. Sounds like a great feature request. PRs welcome!

**Comment by tamird** [comment]
@sydney-runkle is there precedent for this kind of configuration? Do I need to introduce `EmailStrNormalized`?

**Comment by sydney-runkle** [comment]
@tamird,

I think we could probably just introduce an extra property on the `EmailStr` class...

**Comment by tamird** [comment]
How would that work? Is there precedent for such configurability?

**Comment by sydney-runkle** [comment]
@tamird,

Yeah, `EmailStrNormalized` could work, though I'm not sure if it's a high enough demand type that it belongs in Pydantic's code. Maybe `pydantic-extra-types` or simply user code...
